packages/ui/components/ui/tx/view/swap/index.tsx

```diff
@@ -58,7 +62,24 @@ export const SwapViewComponent = ({ value }: { value: SwapView }) => {
   }
 
   if (value.swapView.case === 'opaque') {
-    return <UnimplementedView label='Swap' />;
+    const oneWaySwap = isOneWaySwapOpaque(value) ? getOneWaySwapValuesOpaque(value) : undefined;
+    if (!isOneWaySwapOpaque(value)) {
+      throw new Error(
+        'Attempted to get one-way swap values from a two-way swap. `getOneWaySwapValues()` should only be called with a `SwapView` containing a one-way swap -- that is, a swap with at least one `swapPlaintext.delta*` that has an amount equal to zero.',
+      );
```

Comment:
Should we throw an error? This will crash the UI. How about we show an `Unimplemented` line of text. Looks like you are doing that here? 
https://github.com/penumbra-zone/web/pull/1228/files#diff-281812fff2a966e48bcea90f67cb77e3ca74d0d6ba16044ed78f49db1abbc999R78

If so, it doesn't seem we need to throw.

Reply:
good call, removed as it will be correctly caught as you pointed out.